products/storage/api.yaml

```
@@ -780,4 +780,60 @@ objects:
         name: timeUpdated
         api_name: updated
         description: The modification time of the object metadata.
-
+  - !ruby/object:Api::Resource
+    name: 'HmacKey'
+    kind: 'storage#hmacKey'
+    base_url: projects/{{project}}/hmacKeys
+    create_url: projects/{{project}}/hmacKeys?serviceAccountEmail={{serviceAccountEmail}}
+    self_link: projects/{{project}}/hmacKeys/{{accessId}}
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      guides:
+        'Official Documentation': 'hhttps://cloud.google.com/storage/docs/authentication/managing-hmackeys'
+      api: 'https://cloud.google.com/storage/docs/json_api/v1/projects/hmacKeys'
+    description: |
+      The hmacKeys resource represents an HMAC key within Cloud Storage. The resource
+      consists of a secret and HMAC key metadata. HMAC keys can be used as credentials
+      for service accounts.
+    parameters:
+      - !ruby/object:Api::Type::String
+        name: 'serviceAccountEmail'
+        description: The email address of the key's associated service account.
+        input: true
+        required: true
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'secret'
+        output: true
+        description: 'HMAC secret key material.'
+      - !ruby/object:Api::Type::NestedObject
+        name: 'metadata'
+        description: |
+          HMAC key metadata.
+        properties:
+          - !ruby/object:Api::Type::String
+            name: 'accessId'
+            output: true
+            description: 'The access ID of the HMAC Key.'
+          - !ruby/object:Api::Type::String
+            name: 'etag'
+            output: true
+            description: 'HTTP 1.1 Entity tag for the HMAC key.'
+          - !ruby/object:Api::Type::String
+            name: 'id'
+            output: true
+            description: 'The ID of the HMAC key, including the Project ID and the Access ID.'
+          - !ruby/object:Api::Type::Enum
+            name: 'state'
+            description: 'The state of the key. Can be one of ACTIVE, INACTIVE, or DELETED.'
```

```suggestion
            description: 'The state of the key. Can be one of ACTIVE, INACTIVE, or DELETED.'
            default_value: 'ACTIVE'
```

Right now, users will see a diff immediately after creation I think. This should make Terraform handle it correctly.